### PR TITLE
Automated cherry pick of #104039 upstream release 1.22

### PR DESCRIPTION
**Pull request by CatherineF-dev**
#### What type of PR is this?
/kind bug

#### What this PR does / why we need it:
Backport https://github.com/kubernetes/kubernetes/pull/104039 into k8s 1.22

Summary API metrics didn't have pod network metrics when cgroup-driver=systemd

#### Which issue(s) this PR fixes:

Fixes https://github.com/kubernetes/kubernetes/issues/103366

#### Does this PR introduce a user-facing change?

```release-note
Fix regression in 1.21 where memory metrics can be missing from kubelet stats
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by CatherineF-dev** [comment]
cc @bobbypage @ehashman

**Review by ehashman — COMMENTED** [review]
/priority important-soon
/triage accepted
/lgtm

**Review by mrunalp — APPROVED** [review]

**Review by matthyx — APPROVED** [review]
/lgtm

**Comment by CatherineF-dev** [comment]
Hi team, how can it remove "do-not-merge/cherry-pick-not-approved" label?

**Comment by matthyx** [comment]
> Hi team, how can it remove "do-not-merge/cherry-pick-not-approved" label?

it needs a review from [Release Managers](https://github.com/orgs/kubernetes/teams/release-managers)

**Comment by puerco** [comment]
Thank you @CatherineF-dev !
/lgtm

**Pull request merged**
